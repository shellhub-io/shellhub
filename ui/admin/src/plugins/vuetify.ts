// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
