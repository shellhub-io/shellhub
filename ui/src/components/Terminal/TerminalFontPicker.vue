<template>
  <div class="d-flex flex-column align-center pa-4">
    <v-select
      v-model="currentFontFamily"
      @update:model-value="updateFontSettings"
      :items="availableFonts"
      class="w-100"
      label="Font Family"
    />
    <v-number-input
      v-model="currentFontSize"
      @update:model-value="updateFontSettings"
      control-variant="split"
      variant="outlined"
      density="compact"
      :min="8"
      :max="32"
      hide-details
      inset
      class="w-100"
      label="Font Size"
    />
  </div>
</template>

<script setup lang="ts">
import { storeToRefs } from "pinia";
import useTerminalThemeStore from "@/store/modules/terminal_theme";

const emit = defineEmits<{
  "update:fontSettings": [settings: { fontFamily: string; fontSize: number }];
}>();

const terminalThemeStore = useTerminalThemeStore();
const { availableFonts, currentFontFamily, currentFontSize } = storeToRefs(terminalThemeStore);

const updateFontSettings = async () => {
  await terminalThemeStore.setFontSettings(currentFontFamily.value, currentFontSize.value);
  emit("update:fontSettings", {
    fontFamily: currentFontFamily.value,
    fontSize: currentFontSize.value,
  });
};
</script>
