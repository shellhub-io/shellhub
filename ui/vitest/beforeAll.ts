import MockAdapter from "axios-mock-adapter";
import { beforeAll } from "vitest";
import ResizeObserver from "resize-observer-polyfill";
import { systemApi } from "../src/api/http";

const mockedSystemInfo = {
  authentication: {
    local: false,
    saml: true,
  },
  endpoints: {
    api: "localhost:80",
    ssh: "localhost:22",
  },
  setup: true,
  version: "v0.18.3",
};

const systemApiMock = new MockAdapter(systemApi.getAxios());
systemApiMock.onGet("http://localhost:3000/info").reply(200, mockedSystemInfo);

global.CSS = { supports: () => false } as never;
global.ResizeObserver = ResizeObserver;

beforeAll(() => {
  global.CSS = {
    supports: () => false,
    escape: (str: string) => str,
  } as never;
});
