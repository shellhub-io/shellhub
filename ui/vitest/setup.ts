/**
 * Vitest setup function
 */
export async function setup() {
  global.CSS = {
    supports: () => false,
    escape: (str: string) => str,
  } as never;
}

// FAIL LOUDLY on unhandled promise rejections / errors
process.on("unhandledRejection", (reason) => {
  // eslint-disable-next-line no-console
  console.log("FAILED TO HANDLE PROMISE REJECTION");
  throw reason;
});

/**
 * Vitest Teardown function
 */
export async function teardown() {
  console.log("üìù vitest globalTeardown");
}
