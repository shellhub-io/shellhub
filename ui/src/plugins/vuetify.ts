// Styles
import "@mdi/font/css/materialdesignicons.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { VFileUpload, VFileUploadItem } from "vuetify/labs/VFileUpload";
import "../styles/_variables.scss";
import { aliases as faAliases, fa } from "vuetify/iconsets/fa";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#FFFFFF",
    background: "#F5F5F5",
    tabs: "#F5F5F5",
    foreground: "#F5F5F5",
    paymentForm: "#F5F5F5",
    terminal: "#0f1526",
    "v-theme-background": "#FFFFFF",
    "v-theme-surface": "#FFFFFF",
    "v-theme-card": "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
    terminal: "#0f1526",
    "v-theme-background": "#1E2127",
    "v-theme-surface": "#1E2127",
    "v-theme-card": "#22252B",
  },
};

export default createVuetify({
  components: {
    VFileUpload,
    VFileUploadItem,
  },
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
  defaults: {
    global: {

    },
    VTextField: {
      variant: "outlined",
      density: "comfortable",
    },
    VSelect: {
      variant: "outlined",
      density: "comfortable",
    },
    VTextarea: {
      variant: "outlined",
    },
  },
  icons: {
    faAliases,
    sets: {
      fa,
    },
  },
});
