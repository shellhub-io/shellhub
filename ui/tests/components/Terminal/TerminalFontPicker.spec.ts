import { describe, it, expect, vi, beforeEach } from "vitest";
import { mount, VueWrapper, flushPromises } from "@vue/test-utils";
import { createVuetify } from "vuetify";
import { createPinia, setActivePinia } from "pinia";
import TerminalFontPicker from "@/components/Terminal/TerminalFontPicker.vue";
import useTerminalThemeStore from "@/store/modules/terminal_theme";

vi.mock("fontfaceobserver", () => ({
  default: vi.fn().mockImplementation(() => ({
    load: vi.fn().mockResolvedValue(undefined),
  })),
}));

describe("TerminalFontPicker.vue", () => {
  let wrapper: VueWrapper<InstanceType<typeof TerminalFontPicker>>;
  const vuetify = createVuetify();
  setActivePinia(createPinia());
  const terminalThemeStore = useTerminalThemeStore();

  terminalThemeStore.setFontSettings = vi.fn().mockResolvedValue(undefined);
  terminalThemeStore.$patch({
    currentFontFamily: "Monospace",
    currentFontSize: 15,
  });

  beforeEach(async () => {
    wrapper = mount(TerminalFontPicker, {
      global: {
        plugins: [vuetify],
      },
    });
    await flushPromises();
  });

  it("renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("shows current font size value", async () => {
    const fontSizeInput = wrapper.find(".v-number-input input");
    expect((fontSizeInput.element as HTMLInputElement).value).toBe("15");
  });

  it("calls setFontSettings and emits update:fontSettings when font family changes", async () => {
    const fontFamilySelect = wrapper.findComponent({ name: "VSelect" });

    await fontFamilySelect.setValue("Fira Code");
    await flushPromises();

    expect(terminalThemeStore.setFontSettings).toHaveBeenCalledWith("Fira Code", 15);

    const emitted = wrapper.emitted("update:fontSettings");
    expect(emitted).toHaveLength(1);
    expect(emitted?.[0][0]).toEqual({
      fontFamily: "Fira Code",
      fontSize: 15,
    });
  });

  it("calls setFontSettings and emits update:fontSettings when font size changes", async () => {
    const fontSizeInput = wrapper.find(".v-number-input input");

    await fontSizeInput.setValue(18);
    await flushPromises();

    expect(terminalThemeStore.setFontSettings).toHaveBeenCalledWith("Fira Code", 18);

    const emitted = wrapper.emitted("update:fontSettings");
    expect(emitted).toHaveLength(1);
    expect(emitted?.[0][0]).toEqual({
      fontFamily: "Fira Code",
      fontSize: 18,
    });
  });

  it("uses storeToRefs for reactive store properties", async () => {
    terminalThemeStore.$patch({
      currentFontFamily: "Ubuntu Mono",
      currentFontSize: 22,
    });
    await flushPromises();

    const fontSelect = wrapper.find(".v-select input");
    const fontSizeInput = wrapper.find(".v-number-input input");

    expect((fontSelect.element as HTMLInputElement).value).toBe("Ubuntu Mono");
    expect((fontSizeInput.element as HTMLInputElement).value).toBe("22");
  });
});
