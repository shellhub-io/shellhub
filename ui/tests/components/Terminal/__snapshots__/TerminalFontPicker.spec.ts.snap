// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TerminalFontPicker.vue > renders the component 1`] = `
"<div class="d-flex flex-column align-center pa-4">
  <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-select v-select--single v-select--selected w-100">
    <!---->
    <div class="v-input__control">
      <div class="v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--variant-filled v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-controls="menu-v-0">
        <div class="v-field__overlay"></div>
        <div class="v-field__loader">
          <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
            <!---->
            <div class="v-progress-linear__background"></div>
            <div class="v-progress-linear__buffer" style="width: 0%;"></div>
            <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" for="input-v-2" aria-hidden="false">
            <!---->Font Family
          </label><label class="v-label v-field-label" for="input-v-2">
            <!---->Font Family
          </label>
          <!---->
          <div class="v-field__input" data-no-activator="">
            <!---->
            <!---->
            <div class="v-select__selection"><span class="v-select__selection-text">Monospace<!----></span></div><input size="1" role="combobox" type="text" id="input-v-2" aria-describedby="input-v-2-messages" inputmode="none" aria-expanded="false" aria-controls="menu-v-0" aria-label="Open" title="Open" value="Monospace">
          </div>
          <!---->
        </div>
        <!---->
        <div class="v-field__append-inner">
          <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon" aria-hidden="true"></i>
          <!---->
        </div>
        <div class="v-field__outline">
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <div id="input-v-2-messages" class="v-input__details" role="alert" aria-live="polite">
      <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages">
        <!---->
      </transition-group-stub>
      <!---->
    </div>
  </div>
  <div class="v-input v-input--horizontal v-input--center-affix v-input--density-compact v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-number-input v-number-input--inset v-number-input--split w-100">
    <!---->
    <div class="v-input__control">
      <div class="v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--prepended v-field--variant-outlined v-theme--light v-locale--is-ltr">
        <div class="v-field__overlay"></div>
        <div class="v-field__loader">
          <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
            <!---->
            <div class="v-progress-linear__background"></div>
            <div class="v-progress-linear__buffer" style="width: 0%;"></div>
            <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <div class="v-field__prepend-inner">
          <!---->
          <div class="v-number-input__control"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" style="height: 100%;" tabindex="-1" aria-hidden="true" data-testid="decrement"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-minus mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!---->
              <!---->
            </button>
            <hr class="v-divider v-divider--vertical v-theme--light" aria-orientation="vertical" role="separator">
          </div>
          <!---->
        </div>
        <div class="v-field__field" data-no-activator="">
          <!----><label class="v-label v-field-label" for="input-v-6">
            <!---->Font Size
          </label>
          <!----><input size="1" type="text" id="input-v-6" inputmode="decimal" class="v-field__input" value="15">
          <!---->
        </div>
        <!---->
        <div class="v-field__append-inner">
          <!---->
          <div class="v-number-input__control">
            <hr class="v-divider v-divider--vertical v-theme--light" aria-orientation="vertical" role="separator"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" style="height: 100%;" tabindex="-1" aria-hidden="true" data-testid="increment"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-plus mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!---->
              <!---->
            </button>
          </div>
          <!---->
        </div>
        <div class="v-field__outline">
          <div class="v-field__outline__start"></div>
          <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" for="input-v-6" aria-hidden="false">
              <!---->Font Size
            </label></div>
          <div class="v-field__outline__end"></div>
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</div>"
`;
